actionpack/lib/action_dispatch/journey/formatter.rb

```diff
@@ -44,6 +44,13 @@ def generate(name, options, path_parameters, parameterize = nil)
             parameterized_parts.delete(key)
           end
 
+          parameterized_parts.delete_if do |key, value|
+            required_parts.exclude?(key) &&
+              !path_parameters.key?(key) &&
```

Comment:
Won't this prevent any path parameter not being removed?

Reply:
I was playing with the possible fix and failed specs became green after adding that but now other spec is failing